### PR TITLE
update process_penalties_and_exits

### LINKED ISSUE #279: process_penalties_and_exits
### Description
Upgrade to 0.1 spec

### Acceptance Criteria
* logic implemented

### PR DESCRIPTION
**Pull request by jrhea**
## PR Description
- updated process_penalities_and_exits to reflect the 0.1 spec
- created min/max functions (with Unit Tests) for UnsignedLongs

## Fixed Issue(s)

resolves #279

### CODE REVIEW

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.primitives.UnsignedLong;
+import net.consensys.cava.junit.BouncyCastleExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(BouncyCastleExtension.class)
+class BeaconStateUtilTest {
```
Comment (akhila-raju) [inline ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java:24]: How about some cases for min of same value, max of same value?

**Review by akhila-raju — COMMENTED** [review]
1 comment

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java (hunk 1 above)
Comment (jrhea) [reply ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java:24]: done

**Review by akhila-raju — APPROVED** [review]
LGTM

**Pull request merged**